internal/tasks/resourcesync.go

```diff
@@ -105,6 +105,15 @@ func (r *ResourceSync) run(ctx context.Context, log logrus.FieldLogger, rs *api.
 		return err
 	}
 
+	// Validate that no fleet names conflict with fleets owned by other ResourceSyncs
+	err = r.validateFleetNameConflicts(ctx, fleets, *owner)
+	if err != nil {
+		api.SetStatusConditionByError(&rs.Status.Conditions, api.ConditionTypeResourceSyncResourceParsed, "success", "fail", err)
+		err = fmt.Errorf("resourcesync/%s: error: %w", *rs.Metadata.Name, err)
+		log.Errorf("%e", err)
```

Comment:
if err is type error then this should be
```suggestion
		log.Error(err)
```